### PR TITLE
Improve developer documentation and add reproducible builds docs

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Better structure the Dependencies and Building sections by adding new subsections.
  - Content can be easily referenced now because of these new sections and they show up in the sidebar.
- Add a new section to cover the reproducible builds

#### Rationale:

- Nicer docs and better covered.
- Relates to #6642

#### Preview

- https://ggbecker-content.readthedocs.io/en/readthedocs-support/manual/developer/02_building_complianceascode.html

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]

File: docs/manual/developer/02_building_complianceascode.md [hunk 1]
```diff
@@ -136,23 +153,41 @@ make -j4 rhel7-profile-playbooks # Ansible Playbooks for all RHEL7 profiles
 make -j4 rhel7  # everything above for RHEL7
 ```
 
-(optional) Configure options before building using a GUI tool:
+### Configuring CMAKE options using GUI
```
Comment (jan-cerny) [inline docs/manual/developer/02_building_complianceascode.md:156]:
```suggestion
### Configuring CMake options using GUI
```

File: docs/manual/developer/02_building_complianceascode.md [hunk 2]
```diff
@@ -136,23 +153,41 @@ make -j4 rhel7-profile-playbooks # Ansible Playbooks for all RHEL7 profiles
...
 
-(optional) Use the `ninja` build system (requires the `ninja-build` package):
+### Reproducible Builds
+
+Set the environment variable `SOURCE_DATE_EPOCH` to generate [reproducible builds](https://reproducible-builds.org/).
+For details about the values and meaning of this variable please check this [source](https://reproducible-builds.org/specs/source-date-epoch/):
+
+```bash
+cd build/
+SOURCE_DATE_EPOCH=1614699939 make -j4
```
Comment (jan-cerny) [inline docs/manual/developer/02_building_complianceascode.md:173]: Are there any implications of using parallel builds on the reproducibility?

Comment (ggbecker) [reply docs/manual/developer/02_building_complianceascode.md:173]: I don't think so. It is just a copy and paste matter. I'll remove it.

File: docs/manual/developer/02_building_complianceascode.md (hunk 1 above)
Comment (ggbecker) [reply docs/manual/developer/02_building_complianceascode.md:156]: Ok

**Comment by jan-cerny** [comment]
This is a documentation only change so the OpenShift CI results aren't relevant.

**Pull request merged**